Improve messaging to display clear warning when missing snogrb file

KateFriedman-NOAA opened the pull request:
# Description

This PR resolves bugzilla [1373](http://www2.spa.ncep.noaa.gov/bugzilla/show_bug.cgi?id=1373) (Display clear warning when missing snogrb file). The following changes are included in this PR to resolve the bugzilla:

- add "INFO" message when final snogrb file is decided in script
- add "FATAL ERROR" message when decided-upon snogrb file is missing, `exit` if so

Also included in this PR are a related bug fix uncovered during work to resolve bugzilla and needed COMIN/COMOUT updates, which are part of a larger initiative. Those changes are:

- fix bug with `res` variable setting in `scripts/exgdas_enkf_sfc.sh`
- COMIN/COMOUT updates for the sfcanl jobs

Also included in this PR is the removal of `getges.sh` from the related scripts and removal of `GETGES` mention in some wave job scripts. Full removal of `getges.sh` will be done in a separate PR.

Refs #2451
Refs #1041
Resolves #2500

# Type of change

Bugzilla resolution and bug fix

# Change characteristics

Error handling and messaging updates in related workflow scripts.

# How has this been tested?

Ran C96_atm3DVar CI test on Hercules.

KateFriedman-NOAA committed:
Improve messaging for snogrb setting

- Add INFO echo to state which snogrb file will be used in log
- Also fix bug with res setting in exgdas_enkf_sfc.sh

Resolves bugzilla 1373

Refs #2500

KateFriedman-NOAA committed:
Add FATAL ERROR exit and update COM variables

- Add "FATAL ERROR" message if either snogrb file is missing.
- Update COMIN/COMOUT variable settings (needed)

Refs #2500

KateFriedman-NOAA committed:
Move new INFO message inside if-block

Combine the INFO and FATAL ERROR messaging for snogrb files
into the same if-block.

Refs #2500

CoryMartin-NOAA commented on scripts/exgdas_enkf_sfc.sh:
```diff
@@ -78,18 +78,32 @@ bPDY=${BDATE:0:8}
+# Check if resolution specific FNSNOA exists, if not use t1534 version
+[[ ! -f $FNSNOA ]] && export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f $FNSNOA ]]; then
+  echo "FATAL ERROR: Current snow file ${FNSNOA} is missing. Exiting."
```
```suggestion
  echo "FATAL ERROR: Current cycle snow file ${FNSNOA} is missing. Exiting."
```

KateFriedman-NOAA replied:
Great suggestion, thanks @CoryMartin-NOAA ! Committed at 500aeca7.

CoryMartin-NOAA commented on scripts/exgdas_enkf_sfc.sh:
```diff
@@ -78,18 +78,32 @@ bPDY=${BDATE:0:8}
+  echo "FATAL ERROR: Current snow file ${FNSNOA} is missing. Exiting."
+  exit 1
+else
+  echo "INFO: Current snow file is ${FNSNOA}"
```
```suggestion
  echo "INFO: Current cycle snow file is ${FNSNOA}"
```

CoryMartin-NOAA commented on scripts/exgdas_enkf_sfc.sh:
```diff
@@ -78,18 +78,32 @@ bPDY=${BDATE:0:8}
+# Check if resolution specific FNSNOG exists, if not use t1534 version
+[[ ! -f $FNSNOG ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f $FNSNOG ]]; then
+  echo "FATAL ERROR: Previous snow file ${FNSNOG} is missing. Exiting."
```
```suggestion
  echo "FATAL ERROR: Previous cycle snow file ${FNSNOG} is missing. Exiting."
```

CoryMartin-NOAA commented on scripts/exgdas_enkf_sfc.sh:
```diff
@@ -78,18 +78,32 @@ bPDY=${BDATE:0:8}
+  echo "FATAL ERROR: Previous snow file ${FNSNOG} is missing. Exiting."
+  exit 1
+else
+  echo "INFO: Previous snow file is ${FNSNOG}"
```
```suggestion
  echo "INFO: Previous cycle snow file is ${FNSNOG}"
```

KateFriedman-NOAA committed:
Resolve shellcheck warnings

Refs #2500

CoryMartin-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -62,9 +62,23 @@ LONB_CASE=$((res*4))
+# Check if resolution specific FNSNOA exists, if not use t1534 version
+[[ ! -f $FNSNOA ]] && export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f $FNSNOA ]]; then
+  echo "FATAL ERROR: Current snow file ${FNSNOA} is missing. Exiting."
```
```suggestion
  echo "FATAL ERROR: Current cycle snow file ${FNSNOA} is missing. Exiting."
```

CoryMartin-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -62,9 +62,23 @@ LONB_CASE=$((res*4))
+  echo "FATAL ERROR: Current snow file ${FNSNOA} is missing. Exiting."
+  exit 1
+else
+  echo "INFO: Current snow file is ${FNSNOA}"
```
```suggestion
  echo "INFO: Current cycle snow file is ${FNSNOA}"
```

CoryMartin-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -62,9 +62,23 @@ LONB_CASE=$((res*4))
+# Check if resolution specific FNSNOG exists, if not use t1534 version
+[[ ! -f $FNSNOG ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f $FNSNOG ]]; then
+  echo "FATAL ERROR: Previous snow file ${FNSNOG} is missing. Exiting."
```
```suggestion
  echo "FATAL ERROR: Previous cycle snow file ${FNSNOG} is missing. Exiting."
```

CoryMartin-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -62,9 +62,23 @@ LONB_CASE=$((res*4))
+  echo "FATAL ERROR: Previous snow file ${FNSNOG} is missing. Exiting."
+  exit 1
+else
+  echo "INFO: Previous snow file is ${FNSNOG}"
```
```suggestion
  echo "INFO: Previous cycle snow file is ${FNSNOG}"
```

CoryMartin-NOAA reviewed (commented):
one suggestion in several places just to make things clearer in the logs

KateFriedman-NOAA committed:
Update new snow file messaging to clarify cycle

Refs #2500

WalterKolczynski-NOAA commented on scripts/exgdas_enkf_sfc.sh:
```diff
@@ -78,18 +78,32 @@ bPDY=${BDATE:0:8}
+[[ ! -f ${FNSNOA} ]] && export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f ${FNSNOA} ]]; then
+  echo "FATAL ERROR: Current cycle snow file ${FNSNOA} is missing. Exiting."
+  exit 1
```
If I read George's comments correctly, this isn't supposed to be fatal (yet), as the previous cycle's snow is used.
```suggestion
  echo "INFO: Current cycle snow file ${FNSNOA} is missing. Using previous cycle."
```

WalterKolczynski-NOAA commented on scripts/exgdas_enkf_sfc.sh:
```diff
@@ -78,18 +78,32 @@ bPDY=${BDATE:0:8}
+export FNSNOG=${FNSNOG:-${COMIN_OBS_PREV}/${GPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
+# Check if resolution specific FNSNOG exists, if not use t1534 version
+[[ ! -f ${FNSNOG} ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f ${FNSNOG} ]]; then
+  echo "FATAL ERROR: Previous cycle snow file ${FNSNOG} is missing. Exiting."
```
```suggestion
if [[ ! -f ${FNSNOA} ]] && [[ ! -f ${FNSNOG} ]]; then
  echo "FATAL ERROR: Both current (${FNSNOA}) and previous cycle (${FNSNOG}) snow files are missing."
```

WalterKolczynski-NOAA replied:
@GeorgeGayno-NOAA is this correct, or should it always fail if the snow guess is missing, even if the current snow is present?

WalterKolczynski-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
Same changes as above here.

GeorgeGayno-NOAA commented:
After looking at the script logic more closely, I think there is a way to avoid a FATAL error. Note this logic. New snow is only created once per day (around 00z), but there are snow files every six hours. In most cases, these are simply copies of the 00z file with a new file name. Snow is only applied in the cycling if there is new snow. Hence, the check here on the internal date/time stamp. If FNSNOA and FNSNOG contain the same time stamp, the snow update is shut off by setting FNSNOW="" and CYCLVARS to 99999.  

```
if [[ $(${WGRIB} -4yr "${FNSNOA}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') -le \
  $(${WGRIB} -4yr "${snoprv}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') ]] ; then
  export FNSNOA=" "
  export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
else
```

If FNSNOA or FNSNOG are missing, the same settings could be done and a WARNING message be printed. The script will continue, and the cycling will not update snow.

The snow data is created by an upstream job which has a logic to always populate FNSNOA and FNSNOG files. So I can't envision this ever happening. But if it does, we can continue running and have the SPA investigate the warning.

KateFriedman-NOAA commented:
If I understand the discussion above, we don't want to `FATAL ERROR` and `exit` if either snogrb file is missing, instead we want to print a `WARNING` but continue on. Is that correct? If so:
```
export FNSNOA=${FNSNOA:-${COMIN_OBS}/${OPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
# Check if resolution specific FNSNOA exists, if not use t1534 version
[[ ! -f ${FNSNOA} ]] && export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
if [[ ! -f ${FNSNOA} ]]; then
  echo "WARNING: Current cycle snow file ${FNSNOA} is not available. Job will continue but not update snow."
else
  echo "INFO: Current cycle snow file is ${FNSNOA}"
fi
export FNSNOG=${FNSNOG:-${COMIN_OBS_PREV}/${GPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
# Check if resolution specific FNSNOG exists, if not use t1534 version
[[ ! -f ${FNSNOG} ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
if [[ ! -f ${FNSNOG} ]]; then
  echo "WARNING: Previous cycle snow file ${FNSNOG} is not available. Job will continue but not update snow."
else
  echo "INFO: Previous cycle snow file is ${FNSNOG}"
fi
```

GeorgeGayno-NOAA commented:
> If I understand the discussion above, we don't want to `FATAL ERROR` and `exit` if either snogrb file is missing, instead we want to print a `WARNING` but continue on. Is that correct? If so:
> 
> ```
> export FNSNOA=${FNSNOA:-${COMIN_OBS}/${OPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
> # Check if resolution specific FNSNOA exists, if not use t1534 version
> [[ ! -f ${FNSNOA} ]] && export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
> if [[ ! -f ${FNSNOA} ]]; then
>   echo "WARNING: Current cycle snow file ${FNSNOA} is not available. Job will continue but not update snow."
> else
>   echo "INFO: Current cycle snow file is ${FNSNOA}"
> fi
> export FNSNOG=${FNSNOG:-${COMIN_OBS_PREV}/${GPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
> # Check if resolution specific FNSNOG exists, if not use t1534 version
> [[ ! -f ${FNSNOG} ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
> if [[ ! -f ${FNSNOG} ]]; then
>   echo "WARNING: Previous cycle snow file ${FNSNOG} is not available. Job will continue but not update snow."
> else
>   echo "INFO: Previous cycle snow file is ${FNSNOG}"
> fi
> ```

Yes, print a warning, but these variables also must be set:

```
 export FNSNOA=" "
 export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
```

KateFriedman-NOAA commented:
> export FNSNOA=" "
>  export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"

Gotcha. Set those for both the missing current cycle and previous cycle snogrb files? Should these if-blocks get merged for any reason?

Like:
```
export FNSNOA=${FNSNOA:-${COMIN_OBS}/${OPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
# Check if resolution specific FNSNOA exists, if not use t1534 version
[[ ! -f ${FNSNOA} ]] && export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
if [[ ! -f ${FNSNOA} ]]; then
  echo "WARNING: Current cycle snow file ${FNSNOA} is not available. Job will continue but not update snow."
  export FNSNOA=" "
  export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
else
  echo "INFO: Current cycle snow file is ${FNSNOA}"
fi
export FNSNOG=${FNSNOG:-${COMIN_OBS_PREV}/${GPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
# Check if resolution specific FNSNOG exists, if not use t1534 version
[[ ! -f ${FNSNOG} ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
if [[ ! -f ${FNSNOG} ]]; then
  echo "WARNING: Previous cycle snow file ${FNSNOG} is not available. Job will continue but not update snow."
  export FNSNOA=" "
  export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
else
  echo "INFO: Previous cycle snow file is ${FNSNOG}"
fi
```

GeorgeGayno-NOAA commented:
Something like (not sure the syntax is exactly correct):

```
# one or both files dont exist. turn off cycling of snow.
if [[ -f ${FNSNOA} ]] || [ -f ${snoprv} ]]; then
    export FNSNOA=" "
    export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
# both files exist. check grib date of each file. if the current data it not newer than the previous data, turn off cycling.
elif [ $($WGRIB -4yr $FNSNOA 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') -le \
     $($WGRIB -4yr $snoprv 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') ] ; then
    export FNSNOA=" "
    export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
# current data is newer than previous data. apply snow.
else
    export SNOW_NUDGE_COEFF=${SNOW_NUDGE_COEFF:-0.}
    export CYCLVARS="FSNOL=${SNOW_NUDGE_COEFF},$CYCLVARS"
fi
```

I think that will work. There may be a way to combine the first two branches.  Also, how is $snoprv set when GETGES is called? Can we do a '-f' check on it?

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

KateFriedman-NOAA commented:
> how is $snoprv set when GETGES is called?

When `RUN_GETGES=NO` it uses the value of `FNSNOG`:
```
+ exgdas_enkf_sfc.sh[108]: snoprv=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS_test/COMROOT/C96C48_hybatmDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
```
If `RUN_GETGES=YES` then it would invoke this line:
```
snoprv=$($GETGESSH -q -t snogrb_$JCAP_CASE -e $gesenvir -n $GDUMP -v $GDATE)
```
I tried that by hardcoding `RUN_GETGES=YES` and hit a bunch of unbound variables and issues. Can tell we never invoke `getges.sh` in our v17 testing. I think we should rework this part and/or consider updating `getges.sh` or doing away with it. Thoughts?

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

KateFriedman-NOAA commented:
@GeorgeGayno-NOAA checking in on your thoughts from my last comment. Thanks!

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA checking in on your thoughts from my last comment. Thanks!

If `getges.sh` is not used and if it is not required for operations, then I would get rid of it.

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

KateFriedman-NOAA committed:
Remove reference to getges

Refs #2500

KateFriedman-NOAA committed:
Remove getges and replace snoprv with FNSNOG

Refs #2500

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/bugzilla_1373

* origin/develop:
  Enable HPSS via Globus on Hercules and Orion
  Remove job name from product files & update GEMPAK module. (#3415)
  `link` instead of `copy` in staging jobs (#3410)
  Migrate CI Jenkins to role account on Hera (#3414)
  Add rocotorc documentation when using scrontab (#3417)
  Update jgdas atmos verfozn and verfrad with COMIN/COMOUT prefix instead of COM (#3342)
  Add configuration for empirically-corrected ozone parameters (#3386)
  Enable global-workflow to run C768C384 GSI on Gaea-C6 (#3412)
  Move logical checks into if blocks (#3339)
  Adding Jenkins CI to GaeaC6 using role account (#3389)
  Enable GDASApp g-w CI cases to run on wcoss2 (#3399)
  CI/CD Test on Gaea C5- And update config.gaea under ci/platform (#3280)
  Enable cycling support for Gaea C6 (#3323)
  Update enkf archive jobs to use COMIN/COMOUT (#3393)
  Copy marine ensemble output observation diags and spread (#3407)
  Ci testing on aws 2 (#3408)
  Disable METplus jobs on Hera (#3403)
  Add the mean EnKF soil increment to the deterministic member (#3295)
  Add mpich/8.1.19 to the WCOSS2 LD_LIBRARY_PATH for GDASApp jobs (#3396)
  Change order of RUNs (#3335)
  CI testing on aws (#3391)

JessicaMeixner-NOAA reviewed (dismissed):
Approve these changes  to the wave related files, did not look at others.

KateFriedman-NOAA commented:
> > @GeorgeGayno-NOAA checking in on your thoughts from my last comment. Thanks!
> 
> If `getges.sh` is not used and if it is not required for operations, then I would get rid of it.

@GeorgeGayno-NOAA please see the updated changeset in this PR. I removed the usage of `getges.sh` from the sfc scripts and by doing so I also removed the `snoprv` variable. Let me know if these changes look good to resolve the bugzilla. Thanks!

GeorgeGayno-NOAA commented:
@KateFriedman-NOAA - we can avoid a FATAL error and instead issue a warning. On Hera, see my updates to `exglobal_atmos_sfcanl.sh` in `/scratch2/NCEPDEV/stmp1/George.Gayno/global-workflow/scripts`. If that logic works, I can add it to `exgdas_enkf_sfc.sh`.

How can we test this?

DavidHuber-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -63,23 +61,29 @@ LONB_CASE=$((res*4))
+  exit 1
+else
+  echo "INFO: Current cycle snow file is ${FNSNOA}"
 fi
```
```suggestion
if [[ ! -f ${FNSNOA} ]]; then
  export FNSNOA="${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"
fi
if [[ ! -f ${FNSNOA} ]]; then
  echo "WARNING: Current cycle snow file ${FNSNOA} is missing.  Snow coverage will not be updated."
else
  echo "INFO: Current cycle snow file is ${FNSNOA}"
fi
```

GeorgeGayno-NOAA replied:
Looks good.

KateFriedman-NOAA replied:
I applied the suggested changes @ 135930b49, thanks guys!

DavidHuber-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -63,23 +61,29 @@ LONB_CASE=$((res*4))
+export FNSNOG=${FNSNOG:-${COMIN_OBS_PREV}/${GPREFIX}snogrb_t${JCAP_CASE}.${LONB_CASE}.${LATB_CASE}}
+# Check if resolution specific FNSNOG exists, if not use t1534 version
+[[ ! -f ${FNSNOG} ]] && export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
+if [[ ! -f ${FNSNOG} ]]; then
+  echo "FATAL ERROR: Previous cycle snow file ${FNSNOG} is missing. Exiting."
+  exit 1
 else
-  snoprv=${snoprv:-${FNSNOG}}
+  echo "INFO: Previous cycle snow file is ${FNSNOG}"
 fi
```
```suggestion
if [[ ! -f ${FNSNOG} ]]; then
  export FNSNOG="${COMIN_OBS_PREV}/${GPREFIX}snogrb_t1534.3072.1536"
fi

if [[ ! -f ${FNSNOG} ]]; then
    echo "WARNING: Previous cycle snow file ${FNSNOG} is missing.  Snow coverage will not be updated."
else
  snoprv=${snoprv:-${FNSNOG}}
  echo "INFO: Previous cycle snow file is ${FNSNOG}"
fi
```

GeorgeGayno-NOAA replied:
I thought Kate removed `snoprv`. I made my mods with 6de1f91, which did not have that variable.

KateFriedman-NOAA replied:
I applied the suggested changes manually @ 135930b49. I indeed removed `snoprv`.

DavidHuber-NOAA commented on scripts/exglobal_atmos_sfcanl.sh:
```diff
@@ -63,23 +61,29 @@ LONB_CASE=$((res*4))
+# Set CYCLVARS by checking grib date of current snogrb vs that of prev cycle
 if [[ $(${WGRIB} -4yr "${FNSNOA}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') -le \
-  $(${WGRIB} -4yr "${snoprv}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') ]] ; then
+  $(${WGRIB} -4yr "${FNSNOG}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') ]] ; then
```
```suggestion
# if any snow files are missing, don't apply snow in the global_cycle step.
if [[ ! -f ${FNSNOA} ]] || [[ ! -f ${FNSNOG} ]]; then
  export FNSNOA=" "
  export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
# Set CYCLVARS by checking grib date of current snogrb vs that of prev cycle
elif [[ $(${WGRIB} -4yr "${FNSNOA}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') -le \
  $(${WGRIB} -4yr "${FNSNOG}" 2>/dev/null | grep -i snowc | awk -F: '{print $3}' | awk -F= '{print $2}') ]] ; then
  export FNSNOA=" "
  export CYCLVARS="FSNOL=99999.,FSNOS=99999.,"
```

KateFriedman-NOAA replied:
I applied the suggested changes manually @ https://github.com/NOAA-EMC/global-workflow/commit/135930b4966ea6fd2dfac2c97a341a12449f390b

DavidHuber-NOAA reviewed (changes requested):
Looking through @GeorgeGayno-NOAA's suggestions in `/scratch2/NCEPDEV/stmp1/George.Gayno/global-workflow/scripts/exglobal_atmos_sfcanl.sh`, I think I am OK with issuing a warning instead of a FATAL ERROR.  I've added suggestions from his script (please check my suggestions, George).  Thoughts @KateFriedman-NOAA?

Similar changes would need to be added to `exgdas_enkf_sfc.sh`.

DavidHuber-NOAA commented:
@GeorgeGayno-NOAA I think we could test this locally in a one-off test case by adding an `rm -f "${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"` to the beginning of `exglobal_atmos_sfcanl.sh` and `exgdas_enkf_sfc.sh`.  You could then run a C96C48_hybatmaerosnowDA test case.

KateFriedman-NOAA committed:
Change FATAL to WARNING

Refs #2500

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I think we could test this locally in a one-off test case by adding an `rm -f "${COMIN_OBS}/${OPREFIX}snogrb_t1534.3072.1536"` to the beginning of `exglobal_atmos_sfcanl.sh` and `exgdas_enkf_sfc.sh`. You could then run a C96C48_hybatmaerosnowDA test case.

Sounds good. I am happy to look at test results.

KateFriedman-NOAA commented:
@GeorgeGayno-NOAA Please see the following log on Hera:
`/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/logs/2021122018/gfs_sfcanl.log`

I added the suggested file removal from @DavidHuber-NOAA and it worked to show some of the logic. 1) The t1534 snow file was removed from the obs folder, 2) the lower res file isn't available so it set it for the t1534 file, 3) the t1534 file isn't available (since we removed it), so it kicked off the WARNING message.
```
+ exglobal_atmos_sfcanl.sh[56]: JCAP_CASE=190
+ exglobal_atmos_sfcanl.sh[57]: LATB_CASE=192
+ exglobal_atmos_sfcanl.sh[58]: LONB_CASE=384
+ exglobal_atmos_sfcanl.sh[60]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[63]: export 'FNTSFA=        '
+ exglobal_atmos_sfcanl.sh[63]: FNTSFA='        '
+ exglobal_atmos_sfcanl.sh[64]: export FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.seaice.5min.blend.grb
+ exglobal_atmos_sfcanl.sh[64]: FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.seaice.5min.blend.grb
+ exglobal_atmos_sfcanl.sh[65]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[65]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[67]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t190.384.192 ]]
+ exglobal_atmos_sfcanl.sh[68]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[68]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[70]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[71]: echo 'WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.
+ exglobal_atmos_sfcanl.sh[75]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[75]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[77]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t190.384.192 ]]
+ exglobal_atmos_sfcanl.sh[78]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[78]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[80]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[83]: echo 'INFO: Previous cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536'
INFO: Previous cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[87]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 ]]
```

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA Please see the following log on Hera: `/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/logs/2021122018/gfs_sfcanl.log`
> 
> I added the suggested file removal from @DavidHuber-NOAA and it worked to show some of the logic. 1) The t1534 snow file was removed from the obs folder, 2) the lower res file isn't available so it set it for the t1534 file, 3) the t1534 file isn't available (since we removed it), so it kicked off the WARNING message.
> 
> ```
> + exglobal_atmos_sfcanl.sh[56]: JCAP_CASE=190
> + exglobal_atmos_sfcanl.sh[57]: LATB_CASE=192
> + exglobal_atmos_sfcanl.sh[58]: LONB_CASE=384
> + exglobal_atmos_sfcanl.sh[60]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[63]: export 'FNTSFA=        '
> + exglobal_atmos_sfcanl.sh[63]: FNTSFA='        '
> + exglobal_atmos_sfcanl.sh[64]: export FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.seaice.5min.blend.grb
> + exglobal_atmos_sfcanl.sh[64]: FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.seaice.5min.blend.grb
> + exglobal_atmos_sfcanl.sh[65]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[65]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[67]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t190.384.192 ]]
> + exglobal_atmos_sfcanl.sh[68]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[68]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[70]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[71]: echo 'WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
> WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.
> + exglobal_atmos_sfcanl.sh[75]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[75]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[77]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t190.384.192 ]]
> + exglobal_atmos_sfcanl.sh[78]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[78]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[80]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[83]: echo 'INFO: Previous cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536'
> INFO: Previous cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/12/obs/gdas.t12z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[87]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gfs.20211220/18/obs/gfs.t18z.snogrb_t1534.3072.1536 ]]
> ```

The logic is working. Do you want to test the other possible permutations?

KateFriedman-NOAA commented:
> The logic is working. Do you want to test the other possible permutations?

Yep, rerunning with the prior file also missing. Will share log and snippet when run.

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1373

KateFriedman-NOAA commented:
Logic works when both current and previous snow files are missing. See log:
`/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/logs/2021122100/gdas_sfcanl.log`
```
+ exglobal_atmos_sfcanl.sh[60]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[61]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[64]: export 'FNTSFA=        '
+ exglobal_atmos_sfcanl.sh[64]: FNTSFA='        '
+ exglobal_atmos_sfcanl.sh[65]: export FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
+ exglobal_atmos_sfcanl.sh[65]: FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
+ exglobal_atmos_sfcanl.sh[66]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[66]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[68]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192 ]]
+ exglobal_atmos_sfcanl.sh[69]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[69]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[71]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[72]: echo 'WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.+ exglobal_atmos_sfcanl.sh[76]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[76]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[78]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192 ]]
+ exglobal_atmos_sfcanl.sh[79]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[79]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[81]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[82]: echo 'WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.
+ exglobal_atmos_sfcanl.sh[88]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[89]: export 'FNSNOA= '
+ exglobal_atmos_sfcanl.sh[89]: FNSNOA=' '
+ exglobal_atmos_sfcanl.sh[90]: export CYCLVARS=FSNOL=99999.,FSNOS=99999.,
+ exglobal_atmos_sfcanl.sh[90]: CYCLVARS=FSNOL=99999.,FSNOS=99999.,
+ exglobal_atmos_sfcanl.sh[101]: [[ YES == \Y\E\S ]]
```

Any other condition to test?

GeorgeGayno-NOAA commented:
> Logic works when both current and previous snow files are missing. See log: `/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/logs/2021122100/gdas_sfcanl.log`
> 
> ```
> + exglobal_atmos_sfcanl.sh[60]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[61]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[64]: export 'FNTSFA=        '
> + exglobal_atmos_sfcanl.sh[64]: FNTSFA='        '
> + exglobal_atmos_sfcanl.sh[65]: export FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
> + exglobal_atmos_sfcanl.sh[65]: FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
> + exglobal_atmos_sfcanl.sh[66]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[66]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[68]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192 ]]
> + exglobal_atmos_sfcanl.sh[69]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[69]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[71]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[72]: echo 'WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
> WARNING: Current cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.+ exglobal_atmos_sfcanl.sh[76]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[76]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[78]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192 ]]
> + exglobal_atmos_sfcanl.sh[79]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[79]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[81]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[82]: echo 'WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
> WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.
> + exglobal_atmos_sfcanl.sh[88]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[89]: export 'FNSNOA= '
> + exglobal_atmos_sfcanl.sh[89]: FNSNOA=' '
> + exglobal_atmos_sfcanl.sh[90]: export CYCLVARS=FSNOL=99999.,FSNOS=99999.,
> + exglobal_atmos_sfcanl.sh[90]: CYCLVARS=FSNOL=99999.,FSNOS=99999.,
> + exglobal_atmos_sfcanl.sh[101]: [[ YES == \Y\E\S ]]
> ```
> 
> Any other condition to test?

If we want to be thorough, we should test the case when the current data is available, but the previous cycle data is missing.

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/bugzilla_1373

KateFriedman-NOAA commented:
Good thinking...see this rerun with the current file in place but not the prior:
`/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/logs/2021122100/gdas_sfcanl.log`
```
+ exglobal_atmos_sfcanl.sh[61]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[64]: export 'FNTSFA=        '
+ exglobal_atmos_sfcanl.sh[64]: FNTSFA='        '
+ exglobal_atmos_sfcanl.sh[65]: export FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
+ exglobal_atmos_sfcanl.sh[65]: FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
+ exglobal_atmos_sfcanl.sh[66]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[66]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[68]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192 ]]
+ exglobal_atmos_sfcanl.sh[69]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[69]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[71]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[74]: echo 'INFO: Current cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536'
INFO: Current cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[76]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[76]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
+ exglobal_atmos_sfcanl.sh[78]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192 ]]
+ exglobal_atmos_sfcanl.sh[79]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[79]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
+ exglobal_atmos_sfcanl.sh[81]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[82]: echo 'WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.
+ exglobal_atmos_sfcanl.sh[88]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[88]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 ]]
+ exglobal_atmos_sfcanl.sh[89]: export 'FNSNOA= '
+ exglobal_atmos_sfcanl.sh[89]: FNSNOA=' '
+ exglobal_atmos_sfcanl.sh[90]: export CYCLVARS=FSNOL=99999.,FSNOS=99999.,
+ exglobal_atmos_sfcanl.sh[90]: CYCLVARS=FSNOL=99999.,FSNOS=99999.,
+ exglobal_atmos_sfcanl.sh[101]: [[ YES == \Y\E\S ]]
```

GeorgeGayno-NOAA commented:
> Good thinking...see this rerun with the current file in place but not the prior: `/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/logs/2021122100/gdas_sfcanl.log`
> 
> ```
> + exglobal_atmos_sfcanl.sh[61]: rm -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[64]: export 'FNTSFA=        '
> + exglobal_atmos_sfcanl.sh[64]: FNTSFA='        '
> + exglobal_atmos_sfcanl.sh[65]: export FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
> + exglobal_atmos_sfcanl.sh[65]: FNACNA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.seaice.5min.blend.grb
> + exglobal_atmos_sfcanl.sh[66]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[66]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[68]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t190.384.192 ]]
> + exglobal_atmos_sfcanl.sh[69]: export FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[69]: FNSNOA=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[71]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[74]: echo 'INFO: Current cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536'
> INFO: Current cycle snow file is /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[76]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[76]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192
> + exglobal_atmos_sfcanl.sh[78]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t190.384.192 ]]
> + exglobal_atmos_sfcanl.sh[79]: export FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[79]: FNSNOG=/scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536
> + exglobal_atmos_sfcanl.sh[81]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[82]: echo 'WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.'
> WARNING: Previous cycle snow file /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 is missing.  Snow coverage will not be updated.
> + exglobal_atmos_sfcanl.sh[88]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211221/00/obs/gdas.t00z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[88]: [[ ! -f /scratch1/NCEPDEV/stmp4/Kate.Friedman/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA/gdas.20211220/18/obs/gdas.t18z.snogrb_t1534.3072.1536 ]]
> + exglobal_atmos_sfcanl.sh[89]: export 'FNSNOA= '
> + exglobal_atmos_sfcanl.sh[89]: FNSNOA=' '
> + exglobal_atmos_sfcanl.sh[90]: export CYCLVARS=FSNOL=99999.,FSNOS=99999.,
> + exglobal_atmos_sfcanl.sh[90]: CYCLVARS=FSNOL=99999.,FSNOS=99999.,
> + exglobal_atmos_sfcanl.sh[101]: [[ YES == \Y\E\S ]]
> ```

Looks good. The logic is working. So, did we satisfy the bugzilla?

KateFriedman-NOAA commented:
> So, did we satisfy the bugzilla?

Yes, the bugzilla asked us to "produce a clear warning if the file is missing". I believe that has been achieved in this PR.

@GeorgeGayno-NOAA Please proceed with approving if you're good with the changes and we'll update the bugzilla to let NCO know it has been resolved in GFSv17. Thanks for your help with this @GeorgeGayno-NOAA and @DavidHuber-NOAA !

DavidHuber-NOAA reviewed (approved):
Looks good to me!  Thanks for the work on this @KateFriedman-NOAA @GeorgeGayno-NOAA!

DavidHuber-NOAA commented:
I don't think additional testing is required for this PR and it is ready to merge.
